ci-operator/step-registry/cucushift/upgrade/toimage/cucushift-upgrade-toimage-commands.sh

```diff
@@ -0,0 +1,128 @@
+    done
+    if (( wait_time_loop_var >= 5 )); then
+        echo >&2 "Timed out waiting for admin-acks completing, exiting" && return 1
+    fi
+}
+
+# Upgrade the cluster to target release
+function upgrade() {
+    local err current_image
+    
+    if ! err="$(oc adm upgrade --to-image="${target}" 2>&1 >/dev/null)" && [[ $err == *"you must pass --allow-explicit-upgrade to continue"* ]]; then
+        echo "Error msg check on oc adm upgrade --to-image missing --allow-explicit-upgrade PASSED"
+    else
+        echo >&2 "Error msg check on oc adm upgrade --to-image missing --allow-explicit-upgrade FAILED, err: ${err}" && return 1
+    fi
+    
+    current_image="$(oc get clusterversion version -o json|jq -r '.status.history[0].image')"
+    if [[ ${target} == "${current_image}" ]]; then
+        echo >&2 "Upgrade with oc adm upgrade --to-image missing --allow-explicit-upgrade starts but expected NOT start, exiting" && return 1
+    fi
+    
+    if [[ "${FORCE_UPDATE}" == "false" ]]; then
+        oc adm upgrade --to-image="${target}" --allow-explicit-upgrade
+        echo "Upgrading cluster to ${target} gets started..."
+    else       
+        oc adm upgrade --to-image="${target}" --allow-explicit-upgrade --force
+        echo "Force upgrading cluster to ${target} gets started..."       
+    fi
+    return 0
+}
+
+# Monitor the upgrade status
+function check_upgrade_status() {
+    local wait_upgrade="${TIMEOUT}" out avail progress
+    while (( wait_upgrade > 0 )); do
+        echo "oc get clusterversion" && oc get clusterversion
+        out="$(oc get clusterversion --no-headers)"
+        avail="$(echo "${out}" | awk '{print $3}')"
+        progress="$(echo "${out}" | awk '{print $4}')"
+        if [[ ${avail} == "True" && ${progress} == "False" && ${out} == *"Cluster version is ${target_version}" ]]; then
+            echo -e "Upgrade succ\n\n"
+            return 0
+        else
+            sleep 5m
+            (( wait_upgrade -= 5 ))
+        fi        
+    done
+    if (( wait_upgrade <= 0 )); then
+        echo >&2 "Upgrade timeout, exiting" && return 1
+    fi
+}
+
+# Check version, state in history
+function check_history() {
+    local cv version state
+    cv=$(oc get clusterversion/version -o json)
+    version=$(echo "${cv}" | jq -r '.status.history[0].version')
+    state=$(echo "${cv}" | jq -r '.status.history[0].state')
+    if [[ ${version} == "${target_version}" && ${state} == "Completed" ]]; then
+        echo "History check PASSED, cluster is now upgraded to ${target_version}" && return 0
+    else
+        echo >&2 "History check FAILED, cluster upgrade to ${target_version} failed, current version is ${version}, exiting" && return 1
+    fi
+}
+
+# Setup proxy if it's present in the shared dir
+if test -f "${SHARED_DIR}/proxy-conf.sh"
+then
+    # shellcheck disable=SC1091
+    source "${SHARED_DIR}/proxy-conf.sh"
+fi
+
+echo "RELEASE_IMAGE_INITIAL is ${RELEASE_IMAGE_INITIAL}"
```

Comment:
Can you add a line to run `oc adm release info "${RELEASE_IMAGE_INITIAL}"`. This would help with debugging.

Reply:
Sure. Updated.